helix-term/src/commands/typed.rs

```diff
@@ -1739,31 +1739,31 @@ fn reflow(
         return Ok(());
     }
 
+    const DEFAULT_MAX_LEN: usize = 79;
+
     let scrolloff = cx.editor.config().scrolloff;
+    let cfg_text_width = cx.editor.config().text_width;
     let (view, doc) = current!(cx.editor);
 
-    const DEFAULT_MAX_LEN: usize = 79;
-
-    // Find the max line length by checking the following sources in order:
+    // Find the text_width by checking the following sources in order:
     //   - The passed argument in `args`
-    //   - The configured max_line_len for this language in languages.toml
+    //   - The configured text-width for this language in languages.toml
+    //   - The configured text-width in config.toml
     //   - The const default we set above
-    let max_line_len: usize = args
+    let text_width: usize = args
         .get(0)
         .map(|num| num.parse::<usize>())
         .transpose()?
-        .or_else(|| {
-            doc.language_config()
-                .and_then(|config| config.max_line_length)
-        })
+        .or_else(|| doc.language_config().and_then(|config| config.text_width))
+        .or(cfg_text_width)
```

Comment:
I think that would be a pretty bad user experience since the default for the global text-width is `None`  so by default `:reflow` would do nothing. Why are we defaulting to 79 instead of 80 here btw?

Reply:
I would change the default to 80 here as we are breaking configs anyway. Seems like that was just missed in the original PR.
@archseer also braught that up in https://github.com/helix-editor/helix/pull/2423#discussion_r869795725

Reply:
thinking about more, i think it makes more sense to make `text-width` always defined with a default value of 80 instead of having this hardcoded number, and maybe make softwrap ignore `text-width` by default (i was quite surprised by the current behaviour, at first i thought it was a bug)